### PR TITLE
#160597069 Add guest room and bed during request creation

### PR DESCRIPTION
**Pull request by ezrqnkemboi**
#### What does this PR do?

This PR ensure that user enters available room and bed space during new travel request creation.

#### Description of Task to be completed?

As a user, I should be able to add the room and bed space during my new travel request. This will ensure that if I will not get any room available, they will look for alternatives or postpone trip duration. 

#### How should this be manually tested?
- git clone https://github.com/andela/travel_tool_back.git
- git checkout ft-check-available-room-bed-160597069
- yarn install
- make rollback && make migrate
- make seed
- Open postman and provide your token in the authorization header. Enter url POST  for making a request: http://localhost:5000/api/v1/requests/ and make a request using [this JSON Data](https://gist.github.com/ezrqnkemboi/ac49853543e794de425e480fcc15b492)

#### Any background context you want to provide?
Ensure that before creating a request, you should create a guest house using [this PR](https://github.com/andela/travel_tool_back/pull/65)
Another point to note is that, the location in the seeded data is Lagos,Nigeria. What comes from frontend is Lagos, Nigeria or Nairobi, Kenya for the case of Nairobi. 
So, My issue is that if this work is to be demoed with the seeded data, then , the raw for guest house location need to be Lagos, Nigeria or Nairobi, Kenya, so that it will bed will not be available. 

#### What are the relevant pivotal tracker stories?

[#160597069](https://www.pivotaltracker.com/story/show/160597069)

#### Screenshots (if appropriate)

screen shot 2018-10-18 at 20 33 05

#### Questions && Concern:
If, seeded data is used to make a request, then the place for location of guest house need to change to Lagos, Nigeria or Nairobi, Kenya. I have handled the spacing when making a query from the frontend.

### CODE REVIEW

**Comment by samuelkubai** [comment]
@ezrqnkemboi Please resolve these conflicts

**Comment by ezrqnkemboi** [comment]
I have resolved the conflicts.

**Comment by daddychukz** [comment]
@ezrqnkemboi fix failing test

**Comment by ezrqnkemboi** [comment]
@daddychukz I will fix the tests in the morning. It is already midnight.

File: src/middlewares/Validator.js [hunk 1]
```diff
@@ -216,6 +216,32 @@ export default class Validator {
...
+      return res.status(400).json({
+        success: false,
+        message: 'Please fill the details for departure date, gender and location'
+      });
+    }
+    const isValidDeparturedate = moment(departureDate, 'YYYY-MM-DD', true).isValid();
+    const isValidArrivalDate = moment(arrivalDate, 'YYYY-MM-YY', true).isValid();
+
+    if (!isValidDeparturedate || (arrivalDate && !isValidArrivalDate)) {
+      return res.status(400).json({
```
[the remark this answers is not in the document]
Comment (ezrqnkemboi) [reply src/middlewares/Validator.js]: Making rectification on this one.

File: src/middlewares/Validator.js [hunk 2]
```diff
@@ -216,6 +216,32 @@ export default class Validator {
...
+  static validateAvailableRooms(req, res, next) {
+    const {
+      gender,
+      departureDate,
+      location,
+      arrivalDate
+    } = req.query;
+
+    if (!departureDate || !gender || !location) {
+      return res.status(400).json({
```
[the remark this answers is not in the document]
Comment (ezrqnkemboi) [reply src/middlewares/Validator.js]: I have noted this and will be making changes on the same.

File: src/helpers/requests/index.js [hunk 3]
```diff
@@ -208,17 +234,46 @@ export const getTotalCount = (status, statusCount) => {
...
+        attributes: ['bedId'],
+      },
+    ],
+  });
+
+  const bookedBeds = await Promise.all(
+    requestWithTrips.trips.map(trip => {
+      if (!!trip.bedId) {
+        const bed = models.Bed.update(
+          {booked: true},
```
[the remark this answers is not in the document]
Comment (ezrqnkemboi) [reply src/helpers/requests/index.js]: This was actually to mark beds has booked when a user has made a request.

**Pull request merged**